libfreerdp/codec/nsc.c

```diff
@@ -407,7 +407,7 @@ BOOL nsc_process_message(NSC_CONTEXT* context, UINT16 bpp,
 
 	if (!freerdp_image_copy(pDstData, DstFormat, nDstStride, nXDst, nYDst,
 				width, height, context->BitmapData,
-				PIXEL_FORMAT_BGRA32, 0, 0, 0, NULL))
+				PIXEL_FORMAT_BGRX32_VF, 0, 0, 0, NULL))
```

Comment:
I got the color format from the code before the 'color conversion' fix. In previous approach, the code is：

        nsc_process_message(xfc->codecs->nsc, cmd->bpp, cmd->width, cmd->height, cmd->bitmapData, cmd->bitmapDataLength);

        XSetFunction(xfc->display, xfc->gc, GXcopy);
        XSetFillStyle(xfc->display, xfc->gc, FillSolid);

        if (xfc->bitmap_size < (cmd->width * cmd->height * 4))
        {
            xfc->bitmap_size = cmd->width * cmd->height * 4;
            xfc->bitmap_buffer = (BYTE*) _aligned_realloc(xfc->bitmap_buffer, xfc->bitmap_size, 16);

            if (!xfc->bitmap_buffer)
            {
                xf_unlock_x11(xfc, FALSE);
                return FALSE;
            }
        }

        pSrcData = xfc->codecs->nsc->BitmapData;
        pDstData = xfc->bitmap_buffer;

        freerdp_image_copy(pDstData, xfc->format, -1, 0, 0,
                    cmd->width, cmd->height, pSrcData, PIXEL_FORMAT_XRGB32_VF, -1, 0, 0, xfc->palette);

================================================
The old PIXEL_FORMAT_XRGB32_VF should corresponds to your new PIXEL_FORMAT_BGRX32_VF